sdk/identity/azure-identity/azure/identity/_credentials/azure_arc.py

```diff
@@ -0,0 +1,138 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+import os
+import time
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.transport import HttpRequest, HttpResponse
+from azure.core.pipeline.policies import (
+    DistributedTracingPolicy,
+    HeadersPolicy,
+    HttpLoggingPolicy,
+    HTTPPolicy,
+    UserAgentPolicy,
+    NetworkTraceLoggingPolicy,
+)
+
+from .. import CredentialUnavailableError
+from .._constants import EnvironmentVariables
+from .._internal.managed_identity_client import ManagedIdentityClient, _get_configuration
+from .._internal.get_token_mixin import GetTokenMixin
+from .._internal.user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import,ungrouped-imports
+    from typing import Any, List, Optional, Union
+    from azure.core.configuration import Configuration
+    from azure.core.credentials import AccessToken
+    from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+    PolicyType = Union[HTTPPolicy, SansIOHTTPPolicy]
+
+
+class AzureArcCredential(GetTokenMixin):
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        super(AzureArcCredential, self).__init__()
+
+        url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
+        if not url:
+            # Azure Arc managed identity isn't available in this environment
+            self._client = None
+
+        identity_config = kwargs.pop("_identity_config", None) or {}
+        config = _get_configuration()
+        client_args = dict(
+            kwargs,
+            _identity_config=identity_config,
+            base_headers={"Metadata": "true"},
+            policies=_get_policies(config),
+            request_factory=functools.partial(_get_request, url),
+        )
+
+        self._client = ManagedIdentityClient(**client_args)
```

Comment:
You could use the conventional pattern here, passing specific keyword arguments followed by `**kwargs`.